openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -238,9 +221,17 @@ export function validateIdentifiers(data) {
     else if (data.name === 'lccn') {
         validId = validateLccn(data, dataConfig, label);
     }
+    if (validId === false) return false;
 
-    if (validId) $('#id-errors').hide();
-    return validId;
+    // checking for duplicate identifier entry on all identifier types
+    // expects parsed ids so placed after validate
+    const entries = document.querySelectorAll(`.${data.name}`);
+    if (isIdDupe(entries, data.value) === true) {
+        return error('#id-errors', 'id-value', dataConfig['That ID already exists for this edition.'].replace(/ID/, label));
+    }
```

Comment:
```suggestion
    const entries = document.querySelectorAll(`.${data.name}`);
    if (isIdDupe(entries, data.value) === true) {
        // isbnOverride being set will override the dupe checker, so clear isbnOverride if there's a dupe.
        if (isbnOverride.get()) {isbnOverride.clear()}
        return error('#id-errors', 'id-value', dataConfig['That ID already exists for this edition.'].replace(/ID/, label));
    }
    
    if (validId === false) return false;
```

Good catch on the problem of `isbnOverride` and the way it evades dupe checking.

What about the above as a solution, in conjunction with moving `if (validId === false) return false;` below to the `isIdDupe()` check, and in conjunction with removing the checks in `isFormatValidIsbn10` and `isFormatValidIsbn13`?

Reply:
This seems to work very well and I'm glad we can easily avoid double checking for dupes. Thanks for the suggestion